spec/models/manageiq/providers/openstack/cloud_manager/template_spec.rb

```diff
@@ -1,8 +1,43 @@
 describe ManageIQ::Providers::Openstack::CloudManager::Template do
   let(:ems) { FactoryGirl.create(:ems_openstack_with_authentication) }
+  let(:image_attributes) { {:name => "image", :ram => "1"} }
   let(:template_openstack) { FactoryGirl.create :template_openstack, :ext_management_system => ems, :ems_ref => 'one_id' }
   let(:service) { double }
 
+  context 'when create_image' do
+    before do
+      allow(ExtManagementSystem).to receive(:find).with(ems.id).and_return(ems)
+      allow(ems).to receive(:with_provider_connection).with(:service => 'Compute').and_yield(service)
+      allow(service).to receive(:images).and_return(images)
+    end
+
+    let(:images) { double }
+
+    context 'with correct data' do
+      it 'should create image' do
+        expect(images).to receive(:create).with(image_attributes).and_return(template_openstack).once
```

Comment:
Using service (and its ```create_image``` method) directly would be easier, no need to a "double mock" then.

Nit: I'm not sure if we care about returned ```template_openstack``` data from Fog call, so we don't have to test it then.